packages/core/src/env-set/create-pool-by-env.ts

```diff
@@ -1,25 +1,51 @@
-import {
-  assertEnv,
-  conditional,
-  conditionalString,
-  getEnv,
-  Optional,
-} from '@silverhand/essentials';
+import { assertEnv, conditional, getEnv, Optional } from '@silverhand/essentials';
 import inquirer from 'inquirer';
 import { createPool } from 'slonik';
 import { createInterceptors } from 'slonik-interceptor-preset';
+import { z } from 'zod';
 
-import { createDatabase, createDatabaseCli } from '@/database/seed';
+import { createDatabase, createDatabaseCli, replaceDsnDatabase } from '@/database/seed';
 
 import { appendDotEnv } from './dot-env';
 import { allYes, noInquiry } from './parameters';
 
 const defaultDatabaseUrl = getEnv('DB_URL_DEFAULT', 'postgres://@localhost:5432');
 const defaultDatabaseName = 'logto';
 
+const initDatabase = async (dsn: string): Promise<[string, boolean]> => {
+  try {
+    return [await createDatabase(dsn, defaultDatabaseName), true];
+  } catch (error: unknown) {
+    const result = z.object({ code: z.string() }).safeParse(error);
+
+    // https://www.postgresql.org/docs/12/errcodes-appendix.html
+    const databaseExists = result.success && result.data.code === '42P04';
+
+    if (!databaseExists) {
+      throw error;
+    }
+
+    if (allYes) {
+      return [replaceDsnDatabase(dsn, defaultDatabaseName), false];
+    }
+
+    const useCurrent = await inquirer.prompt({
+      type: 'confirm',
+      name: 'value',
+      message: `A database named "${defaultDatabaseName}" already exists. Would you like to use it without seeding?`,
```

Comment:
Nit: `seeding` sounds a little bit technical. How about `would you like to reuse it without creating a new one?`

Reply:
cool. actually it should be `without filling the initial data?`